streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1834,6 +1833,7 @@ public Map<String, Object> getRestoreConsumerConfigs(final String clientId) {
 
         // Get restore consumer override configs
         final Map<String, Object> restoreConsumerProps = originalsWithPrefix(RESTORE_CONSUMER_PREFIX);
+        checkIfUnexpectedUserSpecifiedConsumerConfig(restoreConsumerProps, NON_CONFIGURABLE_CONSUMER_DEFAULT_CONFIGS);
```

Comment:
Do we need to make this call here? Isn't the check inside `getCommonConsumerConfigs()` enough?

Side note: I just realized that we use `checkIfUnexpectedUserSpecifiedConsumerConfig` also for the producer. Might be good to rename to `checkIfUnexpectedUserSpecifiedClientConfig` ?